matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/database/EventInsertLiveObserver.kt

```diff
@@ -39,65 +38,58 @@ internal class EventInsertLiveObserver @Inject constructor(
 ) :
         RealmLiveEntityObserver<EventInsertEntity>(realmConfiguration) {
 
+    private val lock = Mutex()
+
     override val query = Monarchy.Query {
         it.where(EventInsertEntity::class.java).equalTo(EventInsertEntityFields.CAN_BE_PROCESSED, true)
     }
 
-    private val onResultsChangedFlow = MutableSharedFlow<RealmResults<EventInsertEntity>>()
-
-    init {
-        onResultsChangedFlow
-                .onEach { handleChange(it) }
-                .launchIn(observerScope)
-    }
-
     override fun onChange(results: RealmResults<EventInsertEntity>) {
-        if (!results.isLoaded || results.isEmpty()) {
-            return
-        }
-        observerScope.launch { onResultsChangedFlow.emit(results) }
```

Comment:
do we know the cause of the issue/reproduction steps?

Reply:
I reproduce a lot. Just sent some reactions, if you go back on room list and open again the timeline, there is a high chance you don't see the reactions as the event is not aggregated.
For the exact cause I don't know, but looks like the inner channel gets blocked by the the looper thread...

Reply:
very strange :thinking: 

I haven't been able to reproduce on my side, the flow always seems to emit

| GIF | 
| --- |
![reactions](https://user-images.githubusercontent.com/1848238/173320486-17a6b359-16c2-46a9-be9d-e12f216e2866.gif)

Reply:
(same for me, I cannot repro)